packages/cocos/webpack.config.js

```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+const webpack = require('webpack');
+const TypescriptDeclarationPlugin = require('typescript-declaration-webpack-plugin');
+
+module.exports = {
+  mode: 'production',
+  entry: './src/index.ts',
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/
+      }
+    ]
+  },
+  resolve: {
+    extensions: ['.tsx', '.ts', '.js'],
+    fallback: {
+      buffer: require.resolve('buffer/')
+    }
+  },
+  output: {
+    filename: 'index.ts',
```

Comment:
Is it should be `.ts`, though?

Reply:
This seems to be a limitation of the cocos engine. It is easiest to import modules in ESM format directly using .ts files. Otherwise, conversion is required. You can see more here. https://docs.cocos.com/creator/3.8/manual/en/scripting/modules/spec.html#module-specification